core/db/migrate/20250207104016_add_primary_taxon_to_products.rb

```diff
@@ -0,0 +1,7 @@
+class AddPrimaryTaxonToProducts < ActiveRecord::Migration[7.0]
+  def change
+    add_column :spree_products, :primary_taxon_id, :integer, null: true
+    add_foreign_key :spree_products, :spree_taxons, column: :primary_taxon_id
+    add_index :spree_products, :primary_taxon_id
```

Comment:
This can be simplified

```suggestion
    change_table :spree_products do |t|
      t.add_reference :primary_taxon, {to_table: :spree_taxons}
    end
```

Reply:
Hey Thomas, could you write 3 lines why and how and I put that into the contribution guides.

Reply:
It's part of the ["stick to Rails conventions" part](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#follow-the-coding-conventions) of the contribution guide already. 

This is the idiomatic way of creating a reference: https://guides.rubyonrails.org/active_record_migrations.html#creating-associations